### PR TITLE
Use upp as a submodule

### LINKED ISSUE #905: Use UPP as a submodule
UPP library is updated very frequently, which requires making a UPP release, building it using hpc-stack on all supported platforms and then updating modules in ufs-weather-model. We can just add the upp as a submodule in fv3atm repo and update that submodule whenever we need to update inline post.

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
## Description

This PR adds UPP submodule. It is used in inline post instead of an external upp library.
No change in baseline is expected.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes https://github.com/ufs-community/ufs-weather-model/issues/905

## Testing

How were these changes tested?  Regression tested on Hera.
What compilers / HPCs was it tested with?  Intel.
Are the changes covered by regression tests? Yes. All tests that run inline-post.
Have the ufs-weather-model regression test been run? Yes. On what platform?  Hera.
- Will the code updates change regression test baseline? No.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on https://github.com/NOAA-EMC/UPP/pull/412

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -10,3 +10,6 @@
   path = ccpp/physics
   url = https://github.com/NCAR/ccpp-physics
   branch = main
+[submodule "upp"]
+  path = upp
+  url = https://github.com/DusanJovic-NOAA/UPP
```
Comment (WenMeng-NOAA) [inline .gitmodules]: @DusanJovic-NOAA When UPP being submodule in the authoritative fv3atm, how will an UPP version be checked out?from the latest branch "develop" or tag "XXX"?

Comment (DusanJovic-NOAA) [reply .gitmodules]: Submodules always point to specific commit, which can be a tag but it does not have to be. I guess most of the time, from the develop fv3atm branch we'll point to a commit in UPP develop branch. Production branch will probably point to a commit in UPP production branch.

Comment (WenMeng-NOAA) [reply .gitmodules]: @DusanJovic-NOAA So in that case, I don't need create a tag from UPP repository site. We may check UPP version via hash number of upp directory on atmfv3 repository, right?

Comment (DusanJovic-NOAA) [reply .gitmodules]: Correct. fv3atm 'knows' only about UPP url (optionally branch) and commit hash.

**Review by junwang-noaa — APPROVED** [review]

**Review by WenMeng-NOAA — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**